lessons/module-2/intro-to-destructuring.md

```diff
@@ -0,0 +1,186 @@
+---
+title: "Intro to Destructuring"
+length: 00
+tags: Destructuring, syntax
+---
+## Learning Goals
+Implement destructuring on arrays and objects
+
+## Vocabulary
+`Destructuring` is a feature introduced with ES6. A feature that enables you to break down a structure, like an array or object, and store its components in variables.
+
+`Array destructuring` is a javascript expression that reduces arrays to smaller atoms where the contents of the array can be easily accessed and referenced by variables.
+The reduction may only be to 1 level or to the least level (depending on the depth of the array and how far you want to destructure it). 
+
+
+## Warm Up
+Return to your notebooks and write what do you think these logs will return?
+```js
+colors = ["Red","Pink","Blue","Black"] //Array to be destructured
+
+var [color1, color2] = colors //Destructuring
+
+console.log(color1)
+console.log(color2)
+```
+ 
+
+<section class="answer">
+### The Answer <br>
+This is what happen in the code above. 
+<img class="medium" src="https://github.com/turingschool/front-end-curriculum/blob/f3b5d757b797c33fc08ff93c80c9122f875f2a3a/assets/images/lessons/intro-to-destructuring/intro-to-destructuring-array.png" />
+</section>
+
+<section class="call-to-action">
+### Check It Out
+Let’s now see a few techniques to use destructuring efficiently.
+Return to your group and discuss the difference between these two examples
+
+ 
+#### Example 1:   
+```js
+colors = ["Red","Pink","Blue","Black"] //Array to be destructured
+
+var [,color1,, color2] = colors //Focus on the use of extra commas to skip through elements
+
+console.log(color1)
+console.log(color2)
+```
+#### Example2:
+```js
+colors = ["","Pink","Blue","Black"] //Array to be destructured
+
+var [color1, ...others] = colors //Focus on the use of ... to assign remaining elements to an array
+
+console.log(color1)
+console.log(others)
+```
+</section>
+
+<section class="answer">
+    
+### The Answer   
+    
+Assigning an Array to a variable
+    
+<img class="medium" src="https://github.com/turingschool/front-end-curriculum/blob/dfec040a99f1673538180127f053d7753b3c9a8e/assets/images/lessons/intro-to-destructuring/intro-to-destructuring-array-example1.png"/>   
```

Comment:
Same scenario here, we can use the relative file path and then make sure you are using dashes in the image name.

`<img class="medium" src="../../assets/images/lessons/intro-todestructuring/intro-to-destructuring-array-example1.png"/>   `

Reply:
Hmmmm... I somehow made it work with permalink. I am not sure where you find that intro-todestructuring folder? I am trying to access the images from this folder intro-to-destructuring. I'll trouble shooting this. Thank you